CMakeLists.txt

```diff
@@ -216,6 +232,9 @@ endif()
 # Source is from grpc-repo so don't need PATCH_COMMAND to install
 # source for this target.
 #
+# Use -DCMAKE_INSTALL_LIBDIR:STRING=lib to workaround a build issue on centos/rhel where it defaults to lib64
```

Comment:
Can we automatically detect the library directory like we do that here?
https://github.com/triton-inference-server/client/blob/3712ec3f0426f7132d3759b84683d7d11e9ba683/CMakeLists.txt#L110-L120

Reply:
Hello, I think this should work too, let me test it and update the PR with it if you want.

Reply:
Yes. I think it is better to programmatically detect the platform and do the needful.

Reply:
I think if we do this we will also need to do changes in triton-core (and maybe in other places).
https://github.com/triton-inference-server/core/blob/main/CMakeLists.txt#L283-L287

I think it is easier to just install everything in lib if there is no proper CentOS support but up to you.

Reply:
Should I also open a PR for triton-core ? Are the libs installed in third-party used in other triton components ?

My understanding is that path to libs needs to be updated everywhere they are used if they get installed in lib64.

Reply:
@tanmayv25 what is your opinion on this ? Do you think we should also change the path to those lib (would now be in lib64 instead of lib) in other Triton components ?

Reply:
third-party is also used in triton clients. See here: https://github.com/triton-inference-server/client/blob/main/CMakeLists.txt
I think it would be better to update all the components for centos in Triton. 

@ax-vivien  Let us know if it is expanding your original scope of the changes and we can take over.

Reply:
I can take care of it.

Reply:
Thank you @ax-vivien! I think we need to update the server repo as well: https://github.com/triton-inference-server/server/blob/4f487a08640cfdd2d6bd90d84909eb0902933ada/CMakeLists.txt. I will run the test to verify once the server repo is updated.

Reply:
Here the review for the server repo: https://github.com/triton-inference-server/server/pull/5836.

Note that `opentelemetry-cpp` still has `-DCMAKE_INSTALL_LIBDIR:STRING=lib` flag, I did not remove it as I was not sure why it was there in the first place.

Reply:
Just pushed a last update, I forgot to update grpc lib_dir which is still lib and not lib64 on CentOS. It is all good on my side and ready for testing.

Reply:
For the flag `-DCMAKE_INSTALL_LIBDIR:STRING=lib` used in `opentelemetry-cpp`, I think we can remove the line and [update the path in the server repo](https://github.com/triton-inference-server/server/pull/5836/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR182) or leave it as is. Either way should be fine.

Reply:
I will remove it so it works as the rest

Reply:
Our CI testing looks good. Thanks again for the contribution. Merging all PRs.

Reply:
Thanks a lot ! When can we expect those changes to be released ?

Reply:
The changes will be included in the 23.06 release.